hail/src/main/scala/is/hail/expr/ir/InferPType.scala

```diff
@@ -159,8 +236,9 @@ object InferPType {
       }
       case ArrayMap(a, name, body) => {
         InferPType(a, env)
-        InferPType(body, env.bind(name, a.pType2.asInstanceOf[PArray].elementType.setRequired(false)))
-        coerce[PStreamable](a.pType2).copyStreamable(body.pType2.setRequired(false))
+        InferPType(body, env.bind(name, a.pType2.asInstanceOf[PArray].elementType))
+        // TODO: Any reason to copy here, besides setting setRequired(false), which we've done away with?
+        coerce[PStreamable](a.pType2).copyStreamable(body.pType2, body.pType2.required)
```

Comment:
you mean the `copyStreamable` part? the element type of a.pType2 is the element type of the input array, not the element type of the output array.

Reply:
I meant to remove this comment, figured it out. I meant, why don't we just return body.pType2. I think I figured it out; it's because we need to wrap the thing in PArray, or Stream.

 Thanks!